src/readiness/index.js

```diff
@@ -45,7 +47,8 @@ function GateContext() {
       }
     });
 
-    splits.on(Events.SDK_SPLITS_ARRIVED, () => {
+    splits.on(Events.SDK_SPLITS_ARRIVED, (isSplitKill) => {
+      if(isSplitKill) return;
```

Comment:
Maybe if you just avoid the assignment of `splitStatus = SPLITS_READY` on the splitKill appearance, you could just use SDK_SPLITS_ARRIVED event, as long as you default that parameter to false (or leave it undefined) and propagate it as true when you emit from the local kill handler.

It'll check the state, if ready hasn't fired nothing is gonna happen. If the SDK is ready to emit updates, it'll emit an update. 

One less event, one less code path, and you avoid the need to propagate the isSplitKill only to be emitted along with the new event.

Reply:
great idea